addons/web_editor/static/src/js/editor/snippets.options.js

```diff
@@ -8777,20 +8780,77 @@ registry.many2one = SnippetOptionWidget.extend({
  * Allows to display a warning message on outdated snippets.
  */
 registry.VersionControl = SnippetOptionWidget.extend({
+
+    //--------------------------------------------------------------------------
+    // Options
+    //--------------------------------------------------------------------------
+
+    /**
+     * Replaces an outdated snippet by its new version.
+     */
+    async replaceSnippet() {
+        // Getting the new block version.
+        const snippet = this.$target[0].dataset.snippet;
+        const snippetTagName = this.$target[0].tagName.toLowerCase();
+        const customizePanelEl = this.$el[0].closest(".o_we_customize_panel");
+        const newBlockEl = customizePanelEl.parentElement
+            .querySelector(`${snippetTagName}[data-snippet=${snippet}]`).cloneNode(true);
```

Comment:
Is not there a better way to access the snippet definition (asking the SnippetMenu via a trigger_up for example?)